### PR TITLE
Add a few missing stability attributes

### LINKED ISSUE #21883: "there is no way to use this feature" for std::collections::hash_state::DefaultState
Upgrading Servo is blocked on this. Is there a work around?

```
/home/simon/projects/servo/components/util/cache.rs:29:45: 29:57 error: use of unmarked library feature
/home/simon/projects/servo/components/util/cache.rs:29           entries: HashMap::with_hash_state(DefaultState),
                                                                                                   ^~~~~~~~~~~~
/home/simon/projects/servo/components/util/cache.rs:29:45: 29:57 note: this is either a bug in the library you are using or a bug in the compiler - there is no way to use this feature
/home/simon/projects/servo/components/util/cache.rs:29           entries: HashMap::with_hash_state(DefaultState),
                                                                                                   ^~~~~~~~~~~~
```

### PR DESCRIPTION
**Pull request by chris-morgan**
I’d kind of like to be able to use HashState in AnyMap, which I can’t do without a stability attribute on it. While I was at it I looked around and found a few more missing.

Closes https://github.com/rust-lang/rust/issues/21883.

### CODE REVIEW

File: src/libstd/time/duration.rs [hunk 1]
```diff
@@ -45,6 +45,7 @@ macro_rules! try_opt {
 
 /// ISO 8601 time duration with nanosecond precision.
 /// This also allows for the negative duration; see individual methods for details.
+#[unstable(feature = "std_misc")]
```
Comment (alexcrichton) [inline src/libstd/time/duration.rs]: Adding `#![feature(std_misc)]` seems to already make this type available, did some code you have not compile with the feature gate?

Comment (chris-morgan) [inline src/libstd/time/duration.rs]: It was only HashState that I found myself unable to use; I didn’t try any of the rest. But still, they should have these markers, should they not?

Comment (alexcrichton) [inline src/libstd/time/duration.rs]: Ah this was probably a vicitm of inheritance. An `#[unstable]` attribute is inherited by all children by default. The entire `std::time::duration` module is marked `#[unstable]` (which this inherits), but the `hash_map` module is not unstable and `Hasher` was just erroneously missing a marker.

**Comment by aturon** [comment]
Note: this is currently blocking Servo's rustup. cc @SimonSapin

**Comment by SimonSapin** [comment]
Fixes #21883.

**Pull request merged**